src/Connect.ts

```diff
@@ -48,6 +53,61 @@ export enum Type {
  * Some connection utility functions
  */
 
+/**
+ * Defines the {@link Params.mediaExt} based on the type of parameters and its endpoint.
+ * This method always assigns a value to params.mediaExt, defaulting to an empty string if indeterminable,
+ * allowing detection of whether the function has been applied to the parameters.
+ * @param type The type of parameters to define.
+ * @param params The parameters for which the media extension is to be defined
+ */
+export function defineMediaExt(type: Type, params: Params) {
+    // Fix mediaExt
+    /// remove all the possible '.' prefix
+    if (params.mediaExt) {
+        let i = 0;
+        while (params.mediaExt.charAt(i) === '.') {
```

Comment:
Why not `lastIndexOf`

Reply:
I look, thx!

Reply:
ah, no, here as noted in comment the goal is to sanatize the mediaExt: We want remove the '.' prefix (to prevent if user put '.ext' as extension instead of 'ext').
This code is in fact a `trimStart('.')`
unfortunatly JS don't get a such function, just has a `trimStart()` to trim space, but we can't choice the char to trim.
I note that it could be welcome maybe a such util function `trimStart(char)/trimEnd(char)/trim(char)`?

Reply:
Note that for optimization reason I count in fist the number of '.', to operate an unique `substring` after the loop.
But in same time charAt is expensive too because create a string everytime! :-D 
We should use charCodeAt rather (returns an integer, char code), but anyway, it's more important for a trimStart/trimEnd/trim function this ind of optimization.

Reply:
trim/trimStart/trimEnd customizable and Efficient added!
it's more clear now the code:
```
export function defineMediaExt(type: Type, params: Params) {
    // Fix mediaExt in removing the possible '.' prefix
    if (params.mediaExt) {
        params.mediaExt = Util.trimStart(params.mediaExt, '.');
    }
```